packages/application-shell/src/utils/get-mc-api-url/get-mc-api-url.ts

```diff
@@ -0,0 +1,19 @@
+import { ApplicationWindow } from '@commercetools-frontend/constants';
+
+declare let window: ApplicationWindow;
+
+const getMcApiUrlFromOrigin = () => {
+  const url = new URL(window.origin);
+  const mcApiUrlHost = url.host.replace('mc.', 'mc-api.');
+
+  return `${url.protocol}//${mcApiUrlHost}`;
+};
+
+export default function getMcApiUrl(appEnv = window.app) {
+  const isServedByProxy = JSON.parse(appEnv.servedByProxy);
```

Comment:
I somehow think this is falely typed. The `servedByProxy` is a string on `window.app` on any production app or a boolean when ran locally. It however is typed to always be boolean. /cc @emmenko

Reply:
This is on purpose, as otherwise it gets confusing and we end up with more complexity on the types level.

When we pass the `window.app` to the AppShell, the values get coerced so we end up with a `boolean`. And since we do a type cast anyway when passing the prop to the AppShell, it does not matter much what the original value was.

Reply:
Yes, I know that we coerce the values shallowly. However, what remains on the window itself is untouched. What flows down the application context is coerced. The configuration of the SDK and Apollo however do not consume anything from the application context. Hence not the coerced values. As a result the value here remains string or boolean. 

As a result you're saying here we should not use the "type-safe" variables from the window?

Or Am I missing something? I feel we mean the same again just from different angles.